File: server/tests/fixtures/alpha_rule_triggers/java/arch_entry_point.java
// Should trigger: arch.entry_point
package com.example.api;

import org.springframework.web.bind.annotation.*;

// Spring REST controller - HTTP entry points
@RestController
@RequestMapping("/api")
public class UserController {
    
    @GetMapping("/users")
    public String getUsers() {
        return "[]";
    }
    
    @PostMapping("/users")
    public String createUser(@RequestBody String body) {
        return "{\"created\": true}";
    }
    
    // Main entry point
    public static void main(String[] args) {
        System.out.println("Starting application...");
    }
}

File: server/tests/fixtures/alpha_rule_triggers/javascript/arch_entry_point.js
// Should trigger: arch.entry_point
// Express HTTP endpoint
const express = require('express');

const app = express();

// HTTP GET handler - entry point
app.get('/api/users', (req, res) => {
    res.json({ users: [] });
});

// HTTP POST handler - entry point
app.post('/api/users', (req, res) => {
    res.json({ created: true });
});

// Event listener - entry point
window.addEventListener('load', () => {
    console.log('Page loaded');
});

// Click handler
document.getElementById('btn').onclick = function() {
    console.log('Button clicked');
};
